analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp9.cs

```diff
@@ -76,14 +76,6 @@ public void PatternMatching(object arg)
             }
             o.ToString();       // Noncompliant
         }
-        if (arg is not not null)
-        {
-            arg.ToString();     // Non-compliant FIXME arg is null here
-        }
-        if (!(arg is not null))
-        {
-            arg.ToString();     // Non-compliant FIXME arg is null here
-        }
```

Comment:
Unrelated fix of UTs: here, `arg` is known to be not null because there is `arg.ToString()` above. So condition is always false and invocation is not visited.

Reply:
I did this because of https://github.com/SonarSource/sonar-dotnet/pull/6029#discussion_r958516524
Thanks for moving this to its own methods (I should have done this in the first place).

Reply:
It was not easy to spot